checks/ec2.go

```diff
@@ -0,0 +1,415 @@
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/aws/aws-sdk-go-v2/service/ssm"
+	"github.com/flanksource/canary-checker/api/external"
+	v1 "github.com/flanksource/canary-checker/api/v1"
+	"github.com/flanksource/canary-checker/pkg"
+)
+
+var (
+	prometheusCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "canary_check_ec2_total",
+			Help: "Number of times the ec2checker has run",
+		},
+		[]string{"region"},
+	)
+	prometheusFailCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "canary_check_ec2_failed",
+			Help: "Number of times the ec2checker has failed",
+		},
+		[]string{"region"},
+	)
+	prometheusPassCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "canary_check_ec2_passed",
+			Help: "Number of times the ec2checker has passed",
+		},
+		[]string{"region"},
+	)
+
+	prometheusStartupTime = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "canary_check_ec2_start_time",
+			Help: "ec2 instance startup time",
+		},
+		[]string{"region"},
+	)
+	prometheusTerminateTime = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "canary_check_ec2_terminate_time",
+			Help: "ec2 instance termination time",
+		},
+		[]string{"region"},
+	)
+	prometheusResponseTime = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "canary_check_ec2_response_time",
+			Help: "ec2 instance http response time",
+		},
+		[]string{"region"},
+	)
+)
+
+func init() {
+	prometheus.MustRegister(prometheusCount, prometheusFailCount, prometheusPassCount, prometheusResponseTime, prometheusStartupTime, prometheusTerminateTime)
+}
+
+type EC2Checker struct {
+	kommons *kommons.Client `yaml:"-" json:"-"`
+}
+
+func (c *EC2Checker) SetClient(client *kommons.Client) {
+	c.kommons = client
+}
+
+func (c EC2Checker) GetClient() *kommons.Client {
+	return c.kommons
+}
+
+// Run: Check every entry from config according to Checker interface
+// Returns check result and metrics
+func (c *EC2Checker) Run(canary v1.Canary) []*pkg.CheckResult {
+	var results []*pkg.CheckResult
+	if canary.Spec.EC2.AMI != "" {
+		results = append(results, c.Check(canary, canary.Spec.EC2))
+	}
+	return results
+}
+
+// Type: returns checker type
+func (c *EC2Checker) Type() string {
+	return "ec2"
+}
+
+func (c *EC2Checker) Check(canary v1.Canary, extConfig external.Check) *pkg.CheckResult {
+	check := extConfig.(v1.EC2Check)
+	prometheusCount.WithLabelValues(check.Region).Inc()
+	namespace := canary.Namespace
+
+	kommons := c.GetClient()
+	if kommons == nil {
+		return HandleFail(check, "Kommons client not configured for ec2 checker")
+	}
+	_, ak, err := kommons.GetEnvValue(check.AccessKeyID, namespace)
+	if err != nil {
+		return HandleFail(check, fmt.Sprintf("Could not parse EC2 access key: %v", err))
+	}
+	_, sk, err := kommons.GetEnvValue(check.SecretKey, namespace)
+	if err != nil {
```

Comment:
nitpick: I don't think we should be using abbreviation `ak` and `sk` here. What's wrong with using accessKey and secretKey ?